packages/babel-types/src/index.js

```diff
@@ -399,6 +405,10 @@ export function inherits(child, parent) {
 
   t.inheritsComments(child, parent);
 
+  const pathCache = traverse.cache.path;
+  if (pathCache.has(parent)) {
+    pathCache.set(child, pathCache.get(parent));
+  }
```

Comment:
Not just a performance win leaving this will cause some tests to fail because the node is orphaned and doesn't have an associated nodepath, so it has to create one and by doing so it will not have the objects that were passed at the time of the ast creation (namely hub and file etc) so it will also lead to a failing test